envoy/rds/config_traits.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/pure.h"
+#include "envoy/rds/config.h"
+
+#include "source/common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Rds {
+
+/**
+ * Traits of the protocol specific route configuration and proto.
+ * The generic rds classes will call the methods of this interface
+ * to get information which are not visible for them directly.
+ */
+class ConfigTraits {
+public:
+  virtual ~ConfigTraits() = default;
+
+  /**
+   * Give the full name of the route configuration proto description.
+   * For example 'envoy.config.route.v3.RouteConfiguration'
+   */
+  virtual std::string resourceType() const PURE;
+
+  /**
+   * Create a dummy config object without actual route configuration.
+   * This object will be used before the first valid route configuration is fetched.
+   */
+  virtual ConfigConstSharedPtr createConfig() const PURE;
+
+  /**
+   * Create an empty route configuration proto object.
+   */
+  virtual ProtobufTypes::MessagePtr createProto() const PURE;
```

Comment:
Nit: `createEmptyProto` for consistency with similar places elsewhere.

Reply:
Why do we need both this and `createConfig`?

Reply:
In the original version at the initialization of `RouteConfigUpdateReceiverImpl::route_config_proto_` need to create an empty route config proto.
Also however the proto is passed to the config but `RouteConfigUpdateReceiverImpl` stores both separately.

I will do the renaming.